src/fswAlgorithms/_GeneralModuleFiles/ekfInterface.cpp

```diff
@@ -0,0 +1,379 @@
+    Eigen::VectorXd k3 = ODEfunction(t0 + h/2., X0 + h*k2/2.);
+    Eigen::VectorXd k4 = ODEfunction(t0 + h, X0 + h*k3);
+
+    Eigen::VectorXd X = X0 + 1./6.*h*(k1 + 2.*k2 + 2.*k3 + k4);
+
+    return X;
+}
+
+/*! Set a minimum value (infinite norm, meaning maximal term) of the covariance before switching to Extended KF updates.
+ * This prevents divergence if the initial covariance is high and the state changes too abruptly
+    @param double infiniteNorm
+    @return void
+    */
+void EkfInterface::setMinimumCovarianceNormForEkf(const double infiniteNorm){
+    this->minCovarNorm = infiniteNorm;
+    this->ckfOnlyMode = false;
+}
+
+/*! Get the minimum value of the covariance before switching to Extended KF updates.
+    @return double infiniteNorm
+    */
+double EkfInterface::getMinimumCovarianceNormForEkf() const {
+    return this->minCovarNorm;
+}
+
+/*! Set the filter initial state vector
+    @param Eigen::VectorXd initialStateInput
+    @return void
+    */
+void EkfInterface::setInitialState(const Eigen::VectorXd &initialStateInput){
+    this->stateInitial.resize(initialStateInput.size());
+    this->stateInitial << initialStateInput;
+}
+
+/*! Get the filter initial state vector
+    @return Eigen::VectorXd stateInitial
+    */
+Eigen::VectorXd EkfInterface::getInitialState() const {
+    return this->stateInitial;
+}
+
+/*! Set the filter initial state covariance
+    @param Eigen::MatrixXd initialCovarianceInput
+    @return void
+    */
+void EkfInterface::setInitialCovariance(const Eigen::MatrixXd &initialCovarianceInput){
+    this->covarInitial.resize(initialCovarianceInput.rows(), initialCovarianceInput.cols());
+    this->covarInitial << initialCovarianceInput;
+}
+
+/*! Get the filter initial state covariance
+    @return Eigen::MatrixXd covarInitial
+    */
+Eigen::MatrixXd EkfInterface::getInitialCovariance() const {
+    return this->covarInitial;
+}
+
+/*! Set the filter process noise
+    @param Eigen::MatrixXd processNoiseInput
+    @return void
+    */
+void EkfInterface::setProcessNoise(const Eigen::MatrixXd &processNoiseInput){
+    this->processNoise.resize(processNoiseInput.rows(), processNoiseInput.cols());
+    this->processNoise << processNoiseInput;
+}
+
+/*! Get the filter process noise
+    @return Eigen::MatrixXd processNoise
+    */
+Eigen::MatrixXd EkfInterface::getProcessNoise() const {
+    return this->processNoise;
+}
+
+/*! Set a unit conversion factor, for instance if desirable to solve for a state in km in the filter, but Basilisk's
+ * outside facing interface is in SI
+    @param double conversion
+    */
+void EkfInterface::setUnitConversionFromSItoState(const double conversion){
+    this->unitConversion = conversion;
+}
+
+/*! Get a unit conversion factor, for instance if desirable to solve for a state in km in the filter, but Basilisk's
+ * outside facing interface is in SI
+    @return double unitConversion
+    */
+double EkfInterface::getUnitConversionFromSItoState() const{
```

Comment:
Your function name is more informative than your variable name....

Reply:
I do that on purpose so that the user interface is very clear. What happens inside the class is a bit easier to contain, but I suppose they should both be as clear